### PR TITLE
WIP: initial banner pattern

### PR DESCRIPTION
**Pull request by sturobson**

### CODE REVIEW

**Comment by sturobson** [comment]
screenshot 2018-11-30 at 08 49 52
screenshot 2018-11-30 at 08 51 04
screenshot 2018-11-30 at 08 51 11

File: components/vf-banner/vf-banner.scss [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+.vf-banner--bottom {
+  bottom: 0;
+}
+
+.vf-banner--phase {
+  background-color: #FFFADC;
+}
+
+.vf-banner--gdpr {
+  background-color: set-color(vf-color-gray-dark);
```
Comment (khawkins98) [inline components/vf-banner/vf-banner.scss]: Did you mean `$vf-data-protection-banner-color--background`?
```suggestion
  background-color: $vf-data-protection-banner-color--background;
```

File: components/vf-banner/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "vf-banner.css"
+  ],
+  "test": "echo \"Error: no test specified\" && exit 1",
+  "repo": "https://github.com/visual-framework/vf-core/tree/master/tree/master/components/vf-banner",
+  "bugs": {
+    "url": "https://github.com/visual-framework/vf-core/issues"
+  },
+  "dependencies": {
+    "@visual-framework/vf-sass-config": "^0.0.12"
+  },
```
Comment (khawkins98) [inline components/vf-banner/package.json]:
```suggestion
    "@visual-framework/vf-tag": "^0.0.12"
  },
```

File: components/vf-banner/vf-banner.scss (hunk 1 above)
Comment (sturobson) [reply components/vf-banner/vf-banner.scss]: well - I will do ;) 

need to think about how best to 'theme' them really -- do we have specific use cases or do we have generic light / dark versions? w/ light being the yellow and dark being the dark grey?

File: components/vf-banner/package.json (hunk 2 above)
Comment (sturobson) [reply components/vf-banner/package.json]: yes, yes, yes :)

Comment (khawkins98) [reply components/vf-banner/package.json]: :P

File: components/vf-banner/vf-banner.scss (hunk 1 above)
Comment (khawkins98) [reply components/vf-banner/vf-banner.scss]: ... to me, I think the specific use cases should be EMBL patterns (embl-banner) and more generic use cases for the vf-core "vf-banner--light"

File: components/vf-banner/vf-banner.scss [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+.vf-banner--top {
+  top: 0;
+}
+
+.vf-banner--bottom {
+  bottom: 0;
+}
+
+.vf-banner--phase {
+  background-color: #FFFADC;
```
Comment (khawkins98) [inline components/vf-banner/vf-banner.scss]: How do we want to handle this custom colour? Should we just have some bundled vf-core utility colours?

`vf-color-yelllow-light`?

Comment (sturobson) [reply components/vf-banner/vf-banner.scss]: if it's to become part of the color palette (in any state) we should be:

- defining them as 'global colours' as design tokens
- making decisions to what's primary/secondary/tertiary etc as design tokens
- getting even granular and defining component tokens in the design tokens

I've yet to start this - for now I've just popped the variable at the top of this component file

**Comment by sturobson** [comment]
we might need to revert this one quickly ;)

**Pull request merged**